From: Andrew Morton <akpm@linux-foundation.org>
Subject: __mod_memcg_lruvec_state(): enhance diagnostics
Date: Wed Apr  3 05:42:47 PM PDT 2024

sysbot reports a warning
(https://lkml.kernel.org/r/0000000000007545d00615188a03@google.com).  Add
additional info to perhaps help with this.

Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memcontrol.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/mm/memcontrol.c~__mod_memcg_lruvec_state-enhance-diagnostics
+++ a/mm/memcontrol.c
@@ -861,10 +861,12 @@ static void __mod_memcg_lruvec_state(str
 		case NR_ANON_THPS:
 		case NR_SHMEM_PMDMAPPED:
 		case NR_FILE_PMDMAPPED:
-			WARN_ON_ONCE(!in_task());
+			if (WARN_ON_ONCE(!in_task()))
+				pr_warn("stat item index: %d\n", idx);
 			break;
 		default:
-			VM_WARN_ON_IRQS_ENABLED();
+			if (VM_WARN_ON_IRQS_ENABLED())
+				pr_warn("stat item index: %d\n", idx);
 		}
 	}
 
_
